sdk/storage/azblob/internal/exported/blob_batch.go

```diff
@@ -0,0 +1,271 @@
+}
+
+// buildSubRequest is used for building the sub-request. Example:
+// DELETE /container0/blob0 HTTP/1.1
+// x-ms-date: Thu, 14 Jun 2018 16:46:54 GMT
+// Authorization: SharedKey account:G4jjBXA7LI/RnWKIOQ8i9xH4p76pAQ+4Fs4R1VxasaE=
+// Content-Length: 0
+func buildSubRequest(req *policy.Request) []byte {
+	var batchSubRequest strings.Builder
+	blobPath := req.Raw().URL.Path
+	if len(req.Raw().URL.RawQuery) > 0 {
+		blobPath += "?" + req.Raw().URL.RawQuery
+	}
+
+	batchSubRequest.WriteString(fmt.Sprintf("%s %s %s%s", req.Raw().Method, blobPath, httpVersion, httpNewline))
+
+	for k, v := range req.Raw().Header {
+		if strings.EqualFold(k, shared.HeaderXmsVersion) {
+			continue
+		}
+		if len(v) > 0 {
+			batchSubRequest.WriteString(fmt.Sprintf("%v: %v%v", k, v[0], httpNewline))
+		}
+	}
+
+	batchSubRequest.WriteString(httpNewline)
+	return []byte(batchSubRequest.String())
+}
+
+// CreateBatchRequest creates a new batch request using the sub-requests present in the BlobBatchBuilder.
+//
+// Example of a sub-request in the batch request body:
+//
+//	--batch_357de4f7-6d0b-4e02-8cd2-6361411a9525
+//	Content-Type: application/http
+//	Content-Transfer-Encoding: binary
+//	Content-ID: 0
+//
+//	DELETE /container0/blob0 HTTP/1.1
+//	x-ms-date: Thu, 14 Jun 2018 16:46:54 GMT
+//	Authorization: SharedKey account:G4jjBXA7LI/RnWKIOQ8i9xH4p76pAQ+4Fs4R1VxasaE=
+//	Content-Length: 0
+func CreateBatchRequest(bb *BlobBatchBuilder) ([]byte, string, error) {
+	batchID, err := createBatchID()
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Create a new multipart buffer
+	reqBody := &bytes.Buffer{}
+	writer := multipart.NewWriter(reqBody)
+
+	// Set the boundary
+	err = writer.SetBoundary(batchID)
+	if err != nil {
+		return nil, "", err
+	}
+
+	partHeaders := make(textproto.MIMEHeader)
+	partHeaders["Content-Type"] = []string{"application/http"}
+	partHeaders["Content-Transfer-Encoding"] = []string{"binary"}
+	var partWriter io.Writer
+
+	for i, req := range bb.SubRequests {
+		if bb.AuthPolicy != nil {
+			_, err := bb.AuthPolicy.Do(req)
+			if err != nil && !strings.EqualFold(err.Error(), "no more policies") {
+				if log.Should(EventSubmitBatch) {
+					log.Writef(EventSubmitBatch, "failed to authorize sub-request for %v.\nError: %v", req.Raw().URL.Path, err.Error())
+				}
+				return nil, "", err
+			}
+		}
+
+		partHeaders["Content-ID"] = []string{fmt.Sprintf("%v", i)}
+		partWriter, err = writer.CreatePart(partHeaders)
+		if err != nil {
+			return nil, "", err
+		}
+
+		_, err = partWriter.Write(buildSubRequest(req))
+		if err != nil {
+			return nil, "", err
+		}
+	}
+
+	// Close the multipart writer
+	err = writer.Close()
+	if err != nil {
+		return nil, "", err
+	}
+
+	return reqBody.Bytes(), batchID, nil
+}
+
+// UpdateSubRequestHeaders updates the sub-request headers.
+// Removes x-ms-version header.
+func UpdateSubRequestHeaders(req *policy.Request) {
+	// remove x-ms-version header from the request header
+	for k := range req.Raw().Header {
```

Comment:
The reason why I am doing `range` over `map` is because of the canonicalization of the HTTP headers.
When the request is created, the `x-ms-version` is stored as,
```
req.Raw().Header["x-ms-version"] = []string{"2020-10-02"}
```
So, if we do `req.Header.Get("x-ms-version")`, it will always return empty string because internally it will canonicalize the `x-ms-version` to `X-Ms-Version` for which we don't have an entry in the `Header` map.

Reply:
Oh yes I forgot we changed the codegen to not use `Header.Set()`.